### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by mausch** [deref/exo#302] [General]
Hi and thank you for this project!

I'm looking to make a Nix package for Exo. One of the fields in there is the licence... and the "Deref Community License Agreement" is not on the [list of recognised licences](https://github.com/NixOS/nixpkgs/blob/3b6c3bee9174dfe56fd0e586449457467abe7116/lib/licenses.nix).
It's not on https://opensource.org/licenses/alphabetical or https://spdx.org/licenses/ either so can I get an official statement whether it's FSF free/libre or not?
Otherwise I can still package it as "unfree" which means the binary won't be cached anywhere so it would have to be compiled on every install...

Many thanks

**Comment by brandonbloom** [comment]
Hi Mausch, Thanks for your interest!

The Deref Community License is based on the Confluent Community License, which is in turn based on the Apache 2 license. You can read a bit of background on this license history here: https://www.confluent.io/confluent-community-license-faq/

The only distinction between this license and Apache 2 is the "excluded purpose" clause, which reads:

> Licensee is not granted the right to, and Licensee shall not, exercise the License for an Excluded Purpose.  For purposes of this Agreement, “Excluded Purpose” means making available any software-as-a-service, platform-as-a-service, infrastructure-as-a-service or other similar online service that competes with Deref products or services that provide the Software.

Presently, there is no software-as-a-service component in Exo, but a few are planned. It is possible/likely that those components will not be part of the core of exo, in which case we will have two separate repositories and will relicense the core as an unmodified free/open/libre license, probably Apache 2.0. Until then, we're being a little bit conservative with our licensing plans, preferring to include this "excluded purpose" clause to protect our investment in this project.

I understand that some folks feel strongly about unmodified, pre-approved free & open licensing, so this is something we'll continue to re-evaluate as time goes on.

> I'm looking to make a Nix package for Exo.

We haven't yet packaged exo ourselves except for our simple shell-based installer script. If you are packaging exo, you'll probably want to build with the `managed` tag to disable support for self-upgrades. Grep for `// +build managed` and `// +build !managed` to see what code this toggles on and off. The other build tag that you'll probably want to enable is `bundle`, see [./.github/workflows/perform-release.yaml](./.github/workflows/perform-release.yaml).

cc @kendru who may have additional details (or maybe can put together a quick doc).

**Comment by brandonbloom** [comment]
I've opened #301 to track packaging concerns & am going to move this issue to be a discussion, in case others want to chime in on the licensing topic.

**Comment by brandonbloom** [comment]
We're going to relicense the exo core as plain Apache 2. @BenElgar will take care of this. Thanks!

**Comment by brandonbloom** [comment]
Done for next release: https://github.com/deref/exo/pull/429

**Comment by mausch** [comment]
Great news, thanks. What does "exo core" mean exactly though? i.e. are there any components in this repo that are still under the proprietary licence?

**Comment by brandonbloom** [comment]
Everything in _this_ repository is licensed Apache v2. There will be some _optional_ web services that exo can leverage coming soon. Those will live in other repositories and may-or-may-not be open source. We decided to make this license change now because we've convinced ourselves that we can achieve a good user experience with this split.
